src/bsp/src/esp/src/Factory.cpp

```diff
@@ -0,0 +1,11 @@
+#include "bsp/Factory.h"
+#include "BSP.h"
+
+namespace BspFactory {
```

Comment:
It isn't really a factory since it's not creating it, someone could think it's uses Heap. We had the same thing in the HiveMind. Rename it to BspContainer